code/modules/mob/living/carbon/xenomorph/castes/queen/abilities_queen.dm

```diff
@@ -154,28 +154,8 @@
 	create_shriekwave() //Adds the visual effect. Wom wom wom
 	//stop_momentum(charge_dir) //Screech kills a charge
 
-	for(var/mob/M in view())
-		if(M && M.client)
-			if(isxeno(M))
-				shake_camera(M, 10, 1)
-			else
-				shake_camera(M, 30, 1) //50 deciseconds, SORRY 5 seconds was way too long. 3 seconds now
-
-	for(var/mob/living/carbon/human/H in oview(7, src))
-		var/dist = get_dist(src,H)
-		var/reduction = max(1 - 0.1 * H.protection_aura, 0) //Hold orders will reduce the Halloss; 10% per rank.
-		var/halloss_damage = (max(0,140 - dist * 10)) * reduction //Max 130 beside Queen, 70 at the edge
-		var/stun_duration = max(0,1.1 - dist * 0.1) * reduction //Max 1 beside Queen, 0.4 at the edge.
-
-		if(dist < 8)
-			to_chat(H, "<span class='danger'>An ear-splitting guttural roar tears through your mind and makes your world convulse!</span>")
-			H.stunned += stun_duration
-			H.KnockDown(stun_duration)
-			H.apply_damage(halloss_damage, HALLOSS)
-			if(!H.ear_deaf)
-				H.ear_deaf += stun_duration * 20  //Deafens them temporarily
-			//Perception distorting effects of the psychic scream
-			addtimer(CALLBACK(GLOBAL_PROC, /proc/shake_camera, H, stun_duration * 10, 0.75), 31)
+	for(var/mob/living/L in range(world.view))
+		L.screech_act(src)

```

Comment:
You could check for `stat`. Something like
```
if(L.stat == DEAD)
	continue
L.screech_act(src)
```
That way it wouldn't waste time processing the dead.

Reply:
unless you need to handle stuff that isnt mobs theres no reason to change this from view() or viewers()

Reply:
is there a difference between `view` or `viewers`?

Reply:
Yes.
`range()` is the most efficient among them, and it will return a list of every (movable?) atom in a certain distance, regardless of conditions.

`view()` will return a list of (movable?) atoms visible by the source. It is less efficient, and it will create quirky behaviors, such as screech not affecting mobs that are next to the source but can't be seen by it (inside a locker/tarp for example) and smoke blocking screech.

`viewers()` is the least efficient among them in general, and it would create even quirkier behavior, with someone blindfolding themselves right before the screech happens to become unaffected, for example. It returns a list of mobs that can see the source.

I'd say `range()` is the best option here.